shell/platform/darwin/macos/framework/Source/FLEEngine.mm

```diff
@@ -345,6 +345,49 @@ - (void)sendOnChannel:(nonnull NSString*)channel message:(nullable NSData*)messa
   }
 }
 
+- (void)sendOnChannel:(NSString*)channel
+              message:(NSData* _Nullable)message
+          binaryReply:(FlutterBinaryReply _Nullable)callback {
+  struct Captures {
+    FlutterBinaryReply reply;
+  };
+  auto captures = std::make_unique<Captures>();
+  captures->reply = callback;
+  auto message_reply = [](const uint8_t* data, size_t data_size, void* user_data) {
+    auto captures = reinterpret_cast<Captures*>(user_data);
+    NSData* reply_data = [NSData dataWithBytes:(void*)data length:data_size];
+    captures->reply(reply_data);
```

Comment:
If |callback| is nil, which the API allows, this will crash. The callback-handling code needs to be wrapped in a nil check.

Reply:
Done